### DISCUSSION TITLE
When use consumer, how to known which one invoke the route?

### DISCUSSION THREAD

**Discussion by huang-x-h** [apache/apisix#7332] [Q&A]
Like the title, we define two consumer (use `key-auth`), then define the route add auth plugin.

Now, when invoke the route, how to known which one consumer? We want to count the consumes invoke routes information, which consumer, inovke which route, invoke how many times.

What i think is modify `key-auth` plugin, when auth success or failed will send a http request to record. Does `apisix` support such hooks?

**Comment by tokers** [comment]
Nope. You need to develop it by yourself.

**Reply by huang-x-h** [reply]
😭 any plans for the feature?

**Reply by tzssangglass** [reply]
The current request hits `route_id ` and `consumer_name` are recorded in the Prometheus plugin.

**Reply by tokers** [reply]
@huang-x-h I think it's easy to support it, such as you can write a piece of code via the `serverless-function` plugin. i.e. You can add a request header `X-APISIX-Consumer` and the value can be fetched from variable `$consumer_name`.

**Reply by huang-x-h** [reply]
> The current request hits `route_id ` and `consumer_name` are recorded in the Prometheus plugin.

Yes, according the test, we can get the information through prometheus or kafka-log

`kafka-log`

![f52b0ec830333d6a43a019eed159bf2](https://user-images.githubusercontent.com/381860/176106822-a5fb6941-7b55-4302-af44-fb6c64e75475.png)

`promethues`

![afdb1ca7826090e5f5faf6b04b1a83f](https://user-images.githubusercontent.com/381860/176106851-5f8d9dc3-652f-443e-a2c2-ba135e263ce9.png)
